### DISCUSSION TITLE
Getting name of a molecule after loading with SDMolSupplier

### DISCUSSION THREAD

**Discussion by amfaber** [rdkit/rdkit#5282] [Q&A] [answered]
Hello,

I want to access the names of my ligands from my sdf file in the rdkit molecule object. Something that seems very straightforward to me, yet I haven't found anyone showing how or any methods that could apply. A simple minimal example is that I load my ligans with
```python
ligs = [mol for mol in Chem.SDMolSupplier("path/to/ligands.sdf")]
```
Now I would like to access the name of the first molecule with something like
```python
lig = ligs[0]
lig.GetName()
```

Is this possible?

I am sure the information isn't discarded by SDMolSupplier(), as when writing the molecules back to an .sdf with SDWriter(), the names are preserved

**Comment by DavidACosgrove** [comment, marked as answer]
Hi,
It should be there as the molecule property '_Name'.  In your case, that
would be accessed with lig.GetProp('_Name').
Dave


On Thu, May 12, 2022 at 10:37 AM amfaber ***@***.***> wrote:

> Hello,
>
> I want to access the names of my ligands from my sdf file in the rdkit
> molecule object. Something that seems very straightforward to me, yet I
> haven't found anyone showing how or any methods that could apply. A simple
> minimal example is that I load my ligans with
>
> ligs = [mol for mol in Chem.SDMolSupplier("path/to/ligands.sdf")]
>
> Now I would like to access the name of the first molecule with something
> like
>
> lig = ligs[0]lig.GetName()
>
> Is this possible?
>
> I am sure the information isn't discarded by SDMolSupplier(), as when
> writing the molecules back to an .sdf with SDWriter(), the names are
> preserved
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5282>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FU5DR7G62ZN72FHP2DVJTGMPANCNFSM5VXUBFSQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by amfaber** [reply]
Thank you! I see the property also shows up if passing includePrivate=True to .GetPropNames( ). Much appreciated :)
